src/plugins/kernel_install/utils.sh

```diff
@@ -51,15 +53,24 @@ function ask_yN()
 #   available kernels in a single line separated by commas. If it gets 0 it
 #   will display each kernel name by line.
 # @prefix Set a base prefix for searching for kernels.
+# @all List all available kernels, not only the ones installed by kw
```

Comment:
I think your PR is the right direction for handling kernels list; for this reason, I would suggest that you keep your new code under this function but create a new one named `list_installed_kernels_based_on_grub` and move the old code to that function. The current implementation is heavily tied to grub which should be treated as a fallback after your change.
Ps.: Maybe create a new commit for that.